Sources/SwiftPlot/PlotStyleHelpers.swift

```diff
@@ -52,18 +52,34 @@ struct Text : Annotation {
     public var color = Color.black
     public var size: Float = 15
     public var location = Point(0.0, 0.0)
+    public var drawBoundingBox: Bool = false
+    public var bboxBorderSize: Float = 5
+    public var bboxColor = Color.white
     public func draw(renderer: Renderer){
+        if drawBoundingBox {
+            var bboxSize = renderer.getTextLayoutSize(text: text, textSize: size)
+            bboxSize.width += 2 * bboxBorderSize
+            bboxSize.height += 2 * bboxBorderSize
+            let bboxRect = Rect(origin: Point(location.x - bboxBorderSize, location.y - bboxBorderSize),
+                            size: bboxSize)
+            renderer.drawSolidRect(bboxRect,
```

Comment:
To me bounding box sounds like a box around the text, should this be `drawRect` instead?

Reply:
I'm not sure, since I was going after the matplotlib implementation of the text annotation that has an argument called  ```set_bbox```.

Reply:
Yes, bounding box is a box around the text. What we want is a background rectangle for the text, and that rectangle can be filled with any color. So, drawSolidRect seems to be the correct function to use here. I am not able to think of a better word for that rectangle. Do you have any ideas?

Reply:
I think that bounding box is fine, but I can change the name to ```backgroundRect``` instead, does that seem better?

Reply:
Should we then have something like `boundingBoxColor` and `boundingBoxStrokeColor`? I haven’t yet looked at how matplotlib does it, I will take a look soon.

Reply:
Ok it seems Matplotlib has a bbox dictionary for specifying an edgecolor and a facecolor. I think we should have a `Box` or `Rectangle` `Annotation` which has a `color`/`foregroundColor` property and a `strokeColor`/`backgroundColor` or with other names. Then there would be a property on the `Text` annotion called something like `boundingBox`, of that type. You would then set the color through that and you can also use the annotation separate from the text if you want.

This `boundingBox` property on `Text` could be optional as well, so now you don't need `drawBackgroundRect: Bool = false`.

Reply:
Also, it seems that our renderers right now can only draw a filled rectangles separate from a stroked rectangle which is not ideal. I think every renderer out there can draw a rectangle with both a fill color and a stroke color at once instead of having to draw two rectangles one on top of the other with different settings. We should probably replace `drawRect` and `drawSolidRect` by a simpler one that accepts a `strokeWidth`, `strokeColor`, `fillColor` and `hatchPattern`.

Reply:
@KarthikRIyer mentioned in #70 

> For boxes, I think drawing a simple rectangle would be of no use in general. We should support drawing a bounding rectangle for a text label. You should be able to get the Size of any text, and you could add some offset to that Size to get the bounding box for that text. For location we can consider the bottom left corner. In the beginning we can just implement a simple version of this but later on we can extend it to support boxes with a different border color, rounded corners, etc.

While a ```Box``` or ```Rectangle``` annotation would be more generalized and universal, I agree that it probably wouldn't have much use other than for text. @odmir what are your thoughts?

Reply:
I did say this, but there's no issue in making it generalized. We can call it Box and make it an optional. I think this might be intuitive to use.

A use case other than text might be something like [this](https://matplotlib.org/3.1.0/gallery/text_labels_and_annotations/demo_annotation_box.html).

> Also, it seems that our renderers right now can only draw a filled rectangle separate from a stroked one which is not ideal.

Yes, I agree that we should have a way to draw a rect with a border in one function. That can be done in another PR. I'll create an issue so that we don't forget.

Reply:
Alright, I'll push some updates to implement this.

Reply:
If I were to implement the ```Box``` annotation type as optional, would I still be able to specify the border size when implementing? I think if it was optional then the location and sizes would be set when declaring the ```Box``` type, similar to the adding ```Annotation``` optionals.

I would like to propose to have have another function, possibly ```drawWithText``` that when given the ```Text``` annotation, border width, size, color, etc. returns the ```Box``` with the calculated sizes and location. That would ultimately make it easier for the end user, who would not have to calculate the locations and sizes to pass themselves.